app/components/edit_channel_info/index.tsx

```diff
@@ -0,0 +1,517 @@
+        height: 40,
+        paddingHorizontal: 15,
+    },
+    loading: {
+        height: 20,
+        width: 20,
+        color: theme.centerChannelBg,
+    },
+    titleContainer30: {
+        flexDirection: 'row',
+        marginTop: 30,
+    },
+    titleContainer15: {
+        flexDirection: 'row',
+        marginTop: 15,
+    },
+    title: {
+        fontSize: 14,
+        color: theme.centerChannelColor,
+        marginLeft: 15,
+    },
+    optional: {
+        color: changeOpacity(theme.centerChannelColor, 0.5),
+        fontSize: 14,
+        marginLeft: 5,
+    },
+    helpText: {
+        fontSize: 14,
+        color: changeOpacity(theme.centerChannelColor, 0.5),
+        marginTop: 10,
+        marginHorizontal: 15,
+    },
+    headerHelpText: {
+        zIndex: -1,
+    },
+    divider: {
+        borderBottomColor: changeOpacity(theme.centerChannelColor, 0.1),
+        borderBottomWidth: 1,
+        marginHorizontal: 15,
+        height: 0,
+    },
+    touchable: {
+        flex: 1,
+        flexDirection: 'row',
+        width: '100%',
+        justifyContent: 'space-between',
+        alignItems: 'flex-start',
+    },
+    touchableText: {
+        flex: 1,
+        flexGrow: 1,
+        fontSize: 16,
+        lineHeight: 24,
+        color: '#3d3c40',
+        paddingVertical: 10,
+        marginLeft: 15,
+    },
+    touchableIcon: {
+        flex: 1,
+        padding: 10,
+        textAlign: 'right',
+    },
+}));
+
+type Props = {
+        channelType?: string;
+        displayName?: FormInput;
+        editing: boolean;
+        enableRightButton: (enable: boolean) => void;
+        error?: string | object;
+        header?: FormInput;
+        onTypeChange: (type: ChannelType) => void;
+        oldDisplayName?: string;
+        oldHeader?: string;
+        oldPurpose?: string;
+        purpose?: FormInput;
+        saving: boolean;
+        testID?: string;
+        type?: string;
+}
+
+export default function EditChannelInfo({
+    channelType,
+    displayName,
+    editing,
+    enableRightButton,
+    error,
+    header,
+    onTypeChange,
+    oldDisplayName,
+    oldHeader,
+    oldPurpose,
+    purpose,
+    saving,
+    testID,
+    type,
+}: Props) {
+    const theme = useTheme();
+    const intl = useIntl();
+
+    const nameInput = useRef<TextInput>(null);
+    const purposeInput = useRef<TextInput>(null);
+    const headerInput = useRef<TextInput>(null);
+
+    const scrollViewRef = useRef<KeyboardAwareScrollView>();
+
+    // const [keyboardPosition, setKeyBoardPosition] = useState<number>(0);
+    const [keyboardVisible, setKeyBoardVisible] = useState<boolean>(false);
+
+    const [headerHasFocus, setHeaderHasFocus] = useState<boolean>(false);
+    const [headerPosition, setHeaderPosition] = useState<number>();
+
+    const blur = () => {
+        if (nameInput?.current) {
+            nameInput.current.blur();
+        }
+
+        // TODO: uncomment below once the channel URL field is added
```

Comment:
Create a ticket get he design and add this channel URL editing field

Reply:
I really think we should remove this entirely, that comment has more than 3 years and no one has ever asked for it

Reply:
Done! FYI, I am running through the test cases for this PR right now, but it is ready for review. I need to add the tests next.